app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -5,28 +5,30 @@ import android.content.Intent
 import android.os.Bundle
 import android.view.View
 import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
+import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
 import androidx.databinding.DataBindingUtil
 import com.example.sharingang.databinding.ActivityMainBinding
 import com.example.sharingang.items.Item
 import com.example.sharingang.items.ItemsViewModel
 
 class MainActivity : AppCompatActivity() {
-    private lateinit var binding: ActivityMainBinding
+    private val viewModel: ItemsViewModel by viewModels()
 
```

Comment:
Could you please explain in a few words why the choice of ViewModels and how it works in our case?

Reply:
One advantage of a ViewModel is that the logic can be extracted from the activities / fragments, and tested independently. Furthemore, you can have a view model encapsulate logic such as accessing a database, which is going to be useful pretty soon. You can also share a viewmodel between fragments, which is convenient.

In our case, this isn't strictly necessary, but since we have a list of items, it's already a bit of a good idea to encapsulate that list into a view model, which just provides a method to add items, and a way to view the current items.